addons/edi/models/edi_record.py

```diff
@@ -292,7 +286,10 @@ def prepare(self, doc, vlist):
         # Create records
         with self.statistics() as stats:
             try:
-                self.create(list(self._add_doc_id(doc, vlist)))
+                for record_vals in vlist:
```

Comment:
Why creating inside the loop, i think for performance is better to create similar as it was before.

Reply:
The original code doesn't work properly, there are errors when there are duplicate product lines in the edi record.

Reply:
`total_list` = [loop inside vlsit and update doc_id]
`self.create(total_list)`
`count = len(total_list)`

Reply:
total_list = [{**d,"doc_id": doc.id} for d in vlist]

Reply:
This worked nicely.